modules/lambda-registrator/variables.tf

```diff
@@ -109,3 +109,32 @@ variable "tags" {
   type        = map(string)
   default     = {}
 }
+
+variable "region" {
+  type        = string
+  description = "AWS region to deploy Lambda registrator."
+  default     = "us-east-1"
+}
+
+variable "private_ecr_repo_name" {
+  description = "The name of the repository to republish the ECR image if one exists. If no name is passed, it is assumed that no repository exists and one needs to be created."
```

Comment:
if the user provides `private_ecr_repo_name` then a private repo gets created with that name otherwise a private repo called `consul-lambda-registrator` gets created.

we can keep the default to "" and use consul-lambda-registrator-${random_suffix} to avoid name collisions.

Reply:
The use of `var.private_ecr_repo_name` is dependent upon the value of `var.ecr_image_uri` and they are mutually exclusive. We should clearly document this in the description. The user can either provide `var.ecr_image_uri` or they can provide `var.private_ecr_repo_name`, depending on the behavior that they want.

Another thought on the UX: in order to use the new "pull and republish to public image" method the way that it currently is, I believe the user has to explicitly pass `-var ecr_image_uri="" since it is a required input` . That seems like a less than ideal user experience. What do you think about providing defaults for both `var.ecr_image_uri=""` and `var.private_ecr_repo_name=""` and doing cross-validation to make sure that exactly one of them is set?